.github/workflows/ci.yml

```diff
@@ -24,7 +24,7 @@ jobs:
   # less overhead to keep things in GH actions. See work on building these
   # in GCP with Cloud Build: https://github.com/firezone/firezone/pull/2234
   build-images:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04-firezone
```

Comment:
Can we have that for elixir unit tests too? They take ~4 minutes and we get timeouts in tests when CPU is too busy

Reply:
Yeah, let me bump down the specs from the 64-core to the 16-core. We're already at $50 for that runner just today, which extrapolates to ~$1500 / mo which I think might be a tad too much to invest at this point

Reply:
On second thought, the kotlin and swift builds will still take ~5 minutes, so anything under that probably isn't worth spending $$ on.

We could add a `hashFiles('**/*.ex', '**/.exs')` cache to the elixir unit tests and just skip unit tests if `elixir/` hasn't changed. In theory unit tests should be isolated. Could do the same for other components too.

Reply:
Do you know if those are CPU-bound? The larger runners seems to still have similar IO characteristics as the free ones (SSD iops, memory bandwidth, network bandwidth) so I'm finding that they don't speed things up all that much in many cases.

The QEMU emulated builds are definitely CPU-bound though.

Reply:
Rust compilation usually sees a benefit from 4 to 8 cores for example, anywhere from 10% to 90% increase, but most times just 20% or so:

https://arewefastyet.pages.dev/

Reply:
> We could add a hashFiles('**/*.ex', '**/.exs') cache to the elixir unit tests and just skip unit tests if elixir/ hasn't changed. In theory unit tests should be isolated. Could do the same for other components too.

In theory, we can do that and `mix test --stale` but that doesn't feel safe to me. Better wait for extra minute to be sure.

> Do you know if those are CPU-bound?

Everything about Elixir will be CPU-bound. We don't need 64 cores though, default instance has just 2 - having 4 or 8 will be more than enough.

Reply:
Updated.